clients/src/main/java/org/apache/kafka/clients/admin/ListConsumerGroupOffsetsResult.java

```diff
@@ -17,33 +17,62 @@
 
 package org.apache.kafka.clients.admin;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.clients.admin.internals.CoordinatorKey;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.KafkaFuture;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.annotation.InterfaceStability;
 
 import java.util.Map;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
 
 /**
- * The result of the {@link Admin#listConsumerGroupOffsets(String)} call.
+ * The result of the {@link Admin#listConsumerGroupOffsets(List)} call.
  * <p>
  * The API of this class is evolving, see {@link Admin} for details.
  */
 @InterfaceStability.Evolving
 public class ListConsumerGroupOffsetsResult {
 
-    final KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> future;
+    final Map<CoordinatorKey, KafkaFutureImpl<Map<TopicPartition, OffsetAndMetadata>>> futures;
 
-    ListConsumerGroupOffsetsResult(KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> future) {
-        this.future = future;
+    ListConsumerGroupOffsetsResult(final Map<CoordinatorKey, KafkaFutureImpl<Map<TopicPartition,
+        OffsetAndMetadata>>> futures) {
+        this.futures = futures;
     }
 
     /**
-     * Return a future which yields a map of topic partitions to OffsetAndMetadata objects.
-     * If the group does not have a committed offset for this partition, the corresponding value in the returned map will be null.
+     * Return a future which yields a map of group ids to a map of topic partitions to
+     * OffsetAndMetadata objects. If the group doesn't have a committed offset for a specific
+     * partition, the corresponding value in the returned map for that group id will be null.
      */
-    public KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> partitionsToOffsetAndMetadata() {
```

Comment:
We need to keep this public method and deprecate. Perhaps throw an exception if multiple group ids were specified and retain existing behaviour for single group id.